spec/models/classification_pipeline_spec.rb

```diff
@@ -91,8 +101,37 @@
       data: {"ZZZ" => 1}
     )
 
-    expect do
-      pipeline.process(classification)
-    end.not_to change(FetchClassificationsWorker.jobs, :size).from(0)
+    expect { pipeline.process(classification) }.
+      not_to change(FetchClassificationsWorker.jobs, :size).from(0)
   end
+
+  it 'groups extracts before reduction' do
+    Subject.create! id: 2345
+    Workflow.create! id: 1234, project_id: 1
+
+    # "classroom 1" extracts for subject 2345 in workflow 1234
+    Extract.create! extractor_key: 's', workflow_id: 1234, subject_id: 2345, classification_id: 11111, classification_at: DateTime.now, data: { LN: 1 }
```

Comment:
`create :extract, extractor_key: 's', workflow: workflow, subject: subject`, classification_id: 11111, data: {LN: 1}`

Using factories also means you can leave out attributes that you don't care about (in this case `classification_at`).